BZ-42452: feat: Implement UPI payment redirect handling in BlazeWebView

punyamsingh opened the pull request:
## Jira Ticket
refer [this](https://juspay.atlassian.net/browse/BZ-42452)

## Why?

- Enable seamless UPI payment experience for iOS users by automatically redirecting UPI payment URLs to their respective payment apps  
- Prevent UPI payment links from loading within the webview, which can cause payment failures or poor user experience  
- Support major Indian payment applications to ensure broad compatibility across the payment ecosystem  
- Improve payment conversion rates by providing native app experiences for UPI transactions  
- Fix issues with newer UPI schemes like Paytm merchant payments (paytmmp) that were previously unsupported

## Solution

- Added UPIConfig enum with static schemes Set for supported payment apps (upi, phonepe, tez, gpay, paytm, paytmmp, bhim, amazonpay, mobikwik, freecharge, credpay)
- Implemented isUPIIntentUri() function to validate UPI URLs with required parameters (pa, pn, cu, am)
- Added webView(_:decidePolicyFor:decisionHandler:) navigation delegate method to intercept UPI URLs
- Enhanced WKNavigationDelegate to redirect validated UPI links to external payment apps while allowing normal web navigation
- Prevents UPI payment links from loading within webview, enabling native payment app experience

## How to Test?

1. Load the BlazeSDK integrated app on a test device (physical device recommended)
2. Navigate to a payment method that offers UPI payment options (e.g., Simpl Pay-in-3)
3. Complete the login process for the payment service
4. In the payment flow, select and open a UPI payment method (e.g., GPay)
5. Verify that the external UPI app opens successfully (indicating UPI redirect is working)
6. Test with other available UPI payment options in the flow
7. Ensure the payment process completes successfully in the external app
8. Verify that non-UPI navigation within the webview continues to work normally

## Dev Proof
https://drive.google.com/file/d/1VLFVbjxej8sRWBW_JVU2vcmDHYAM7-Pt/view?usp=sharing

## Priority
High

adarshjuspay commented on BlazeSDK/Classes/BlazeWebView.swift:
```diff
@@ -201,5 +201,26 @@ class BlazeWebView: NSObject, WKScriptMessageHandler {
+        }
+
+        let upiSchemes: Set<String> = [
+            "upi", "phonepe", "tez", "gpay", "paytm", "bhim", "amazonpay", "mobikwik", "freecharge", "credpay"
```
can we add dynamic injection and if not provided fallback to this default set?

punyamsingh commented on BlazeSDK/Classes/BlazeWebView.swift:
```diff
@@ -201,5 +201,26 @@ class BlazeWebView: NSObject, WKScriptMessageHandler {
+
+        if let scheme = url.scheme?.lowercased(), upiSchemes.contains(scheme) {
+            _ = openApp(payload: url.absoluteString)
+            decisionHandler(.cancel)
```
resolved.

punyamsingh commented on BlazeSDK/Classes/BlazeWebView.swift:
```diff
@@ -201,5 +201,26 @@ class BlazeWebView: NSObject, WKScriptMessageHandler {
+
+        let upiSchemes: Set<String> = [
+            "upi", "phonepe", "tez", "gpay", "paytm", "bhim", "amazonpay", "mobikwik", "freecharge", "credpay"
+        ]
```
resolved

punyamsingh committed:
BZ-42452: feat: Implement UPI payment redirect handling in BlazeWebView

- Added isUPIIntentUri() function to validate UPI payment URLs with required parameters (pa, pn, cu, am)
- Supports major UPI payment apps: upi, phonepe, tez, gpay, paytm, paytmmp, bhim, amazonpay, mobikwik, freecharge, credpay
- Implemented webView(_:decidePolicyFor:decisionHandler:) navigation delegate method to intercept UPI URLs
- Enhanced WKNavigationDelegate to redirect validated UPI links to external payment apps while allowing normal web navigation
- Prevents UPI payment links from loading within webview, enabling native payment app experience
